### PR TITLE
Split GMMK Pro PCBs into separate revisions, fix RGB mapping on ISO

### LINKED ISSUE #13524: [Feature Request] GMMK Pro ISO support
Now that the GMMK Pro ISO hit the market, I'd like it to be supported by QMK, too

## Feature Request Type

- [X] Core functionality
- [X] Add-on hardware support (eg. audio, RGB, OLED screen, etc.)
- [X] Alteration (enhancement/optimization) of existing feature(s)
- [ ] New behavior

## Description

The GMMK Pro ISO PCB works very similarly to the ANSI one.
Going by the matrix currently available for the ANSI version, Backslash is ka2 and no longer needed, since it does not exist on the ISO board.
The new keys needed are k23 (KC_NUBS) and k84(KC_NUHS).

Copying the proposed AW20216S RGB implementation into my experimental ISO firmware results in good behavior, except for NUBS and NUHS not being lit. Have not tried the competing one.

I have some codework on my end, but I am very inexperienced with the project and github, so I'd love if someone more experienced could do what glorious should have done 2 weeks ago.

cheers.

### PR DESCRIPTION
**Pull request by Gigahawk**
## Description

RGB mapping is unique between ISO and ANSI GMMK Pro PCBs, split each version into a separate revision for proper support.

@Hatbuster @Giesd @tipok I don't actually have an ISO model to test, on so if you could please confirm this actually works.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #13524

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/pro/ansi/ansi.c [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    { 25,     30,     26,     31,     27,     32, 29,     24 },
+    { 41,     36,     42,     37,     43,     38, 35,     40 },
+    { 46,     89,     47,     34,     48,     72, 78,     45 },
+    { 52,     39,     53,     97,     54,     82, 44,     51 },
+    { 58,     63,     59,     64,     NO_LED, 60, 62,     57 },
+    { 11,     90,     55,     17,     33,     49, NO_LED, 69 },
+    { NO_LED, 85,     93,     61,     96,     66, 50,     56 }
+                                 // clang-format on
+                             },
+                             {
```
Comment (Gigahawk) [inline keyboards/gmmk/pro/ansi/ansi.c:35]: @drashna  Where should the `clang-format` guards be for this?

**Comment by Gigahawk** [comment]
```
☒ Missing keyboards/gmmk/pro/ansi/readme.md
☒ Lint check failed!
linting gmmk/pro/iso
☒ Missing keyboards/gmmk/pro/iso/readme.md
☒ Lint check failed!
```

Are individual readmes actually needed for each folder?

**Comment by zvecr** [comment]
You will want to sort out the keymaps and info.json, as the "common" folder approach wont work as the layout macros are not common.

```
keyboards/gmmk/pro/ansi/info.json
keyboards/gmmk/pro/ansi/keymaps/default/keymap.c
keyboards/gmmk/pro/iso/info.json
keyboards/gmmk/pro/iso/keymaps/default/keymap.c
```

**Comment by Hatbuster** [comment]
@Gigahawk I have pulled your branch onto my local repository (god, I'm bad at git) and switched my keyboard to gmmk/pro/iso instead of gmmk/pro. Built it and flashed it.

On my end the LEDs light up now. Seem to be at least roughly in the right position, too, from what I can tell.
Great stuff, thanks for putting this together!

Can't wait for it to be merged into develop after some cleanup :)

File: keyboards/gmmk/pro/iso/readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Flashing example for this keyboard:
+
+    make gmmk/pro/iso:default:flash
+
+To reset the board into bootloader mode, do one of the following:
+
+* Hold the Reset switch mounted on the bottom side of the PCB while connecting the USB cable
+* Hold the Escape key while connecting the USB cable (also erases persistent settings)
+* Fn+Backslash will reset the board to bootloader mode if you have flashed the default QMK keymap
```
Comment (Hatbuster) [inline keyboards/gmmk/pro/iso/readme.md]: Backslash doesn't exist on the ISO layout. Reset is currently on FN+Backspace.

**Review by zvecr — APPROVED** [review]

File: keyboards/gmmk/pro/ansi/ansi.c (hunk 1 above)
Comment (drashna) [reply keyboards/gmmk/pro/ansi/ansi.c:35]: ideally, right after the if-def

File: keyboards/gmmk/pro/iso/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    ),
+
+    [1] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______
+    ),
```
Comment (drashna) [inline keyboards/gmmk/pro/iso/keymaps/via/keymap.c:62]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by ForsakenRei** [comment]
Just updated to the latest commit. It seems the sideglow LED number changed? 
Used my previous `keymap.c` and my sideglow color goes to some other keys...

**Comment by Gigahawk** [comment]
> Just updated to the latest commit. It seems the sideglow LED number changed? 
> Used my previous ANSI `keymap.c` and my sideglow color goes to some other keys...

Taken from https://github.com/qmk/qmk_firmware/blob/521e24a1b99857ccf8c82782c6b40cb3599aef0f/keyboards/gmmk/pro/pro.c and https://github.com/qmk/qmk_firmware/blob/7a7fe4c16cb3220a724d7ca25409826025ec88ab/keyboards/gmmk/pro/ansi/ansi.c

```
 $ diff -w ansi.c pro.c
 1c1
 < /* Copyright 2021 Gigahawk
 ---
 > /* Copyright 2021 Glorious, LLC <salman@pcgamingrace.com>
 16,17c16
 <
 < #include "ansi.h"
 ---
 > #include "pro.h"
 20d18
 < // clang-format off
 240d237
 < // clang-format on
```

**Comment by ForsakenRei** [comment]
> > Just updated to the latest commit. It seems the sideglow LED number changed?
> > Used my previous ANSI `keymap.c` and my sideglow color goes to some other keys...
> 
> Taken from https://github.com/qmk/qmk_firmware/blob/521e24a1b99857ccf8c82782c6b40cb3599aef0f/keyboards/gmmk/pro/pro.c and https://github.com/qmk/qmk_firmware/blob/7a7fe4c16cb3220a724d7ca25409826025ec88ab/keyboards/gmmk/pro/ansi/ansi.c
> 
> ```
>  $ diff -w ansi.c pro.c
>  1c1
>  < /* Copyright 2021 Gigahawk
>  ---
>  > /* Copyright 2021 Glorious, LLC <salman@pcgamingrace.com>
>  16,17c16
>  <
>  < #include "ansi.h"
>  ---
>  > #include "pro.h"
>  20d18
>  < // clang-format off
>  240d237
>  < // clang-format on
> ```

Thanks. I found the under glow LED nubmers are now from 66-92 with some keycode mixed in it instead of purely 82-97. I will make some changes to my own keymap.

**Comment by Gigahawk** [comment]
You must have been using the Glorious mapping, which was ordered by the keyswitch matrix and then the underglow LEDs slapped on at the end.

The merged mapping on develop follows the order of the actual RGB matrix as defined in [this comment](https://github.com/qmk/qmk_firmware/issues/13524#issuecomment-881312541) with column major axis, following the memory layout of the actual RGB driver chips.

**Comment by ForsakenRei** [comment]
Ah make sense. It's not that hard to change those numbers in keymap either.
Thanks for the help!

**Pull request merged**